src/runtime/util.cpp

```diff
@@ -26,4 +27,68 @@ void parseSlice(BoxedSlice* slice, int size, i64* out_start, i64* out_stop, i64*
     if (ret == -1)
         throwCAPIException();
 }
+
+void sliceIndex(Box* b, int64_t* out) {
```

Comment:
Can we just call `_PyEval_SliceIndex` in places that use this?  or maybe have this method call `_PyEval_SliceIndex` followed by `checkAndThrowCAPIException()` instead of mostly duplicating the body?